### PR TITLE
fix(`zetaclient`): report failed ERC20 transfers from TSS

### PR DESCRIPTION
**Pull request by brewmaster012**
# Description

Report failed ERC20 transfers from TSS. This should not happen, but should be reported. 

Closes: <PD-XXXX>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: zetaclient/evm_client.go [hunk 1]
```diff
@@ -454,6 +454,26 @@ func (ob *EVMChainClient) IsSendOutTxProcessed(sendHash string, nonce uint64, co
 					return true, false, nil
 				}
 			}
+		} else { //
```
Comment (lumtis) [inline zetaclient/evm_client.go:457]:
```suggestion
		} else {
```

**Review by lumtis — APPROVED** [review]
Looks legit to me.

I think `IsSendOutTxProcessed` function is way too complex and should be break down into smaller functions

**Review by kingpinXD — APPROVED** [review]

**Comment by lumtis** [comment]
It appears we need to run `make generate`

File: zetaclient/evm_client.go [hunk 2]
```diff
@@ -454,6 +454,26 @@ func (ob *EVMChainClient) IsSendOutTxProcessed(sendHash string, nonce uint64, co
...
+				transaction.GasPrice(),
+				transaction.Gas(),
+				big.NewInt(0),
+				common.ReceiveStatus_Failed,
+				ob.chain,
+				nonce,
+				common.CoinType_ERC20,
+			)
+			if err != nil {
+				logger.Error().Err(err).Msgf("PostReceiveConfirmation error in WatchTxHashWithTimeout; zeta tx hash %s", zetaTxHash)
```
Comment (ws4charlie) [inline zetaclient/evm_client.go:473]: `zetaTxHash` is probably just an empty string when `if err != nil`

Comment (brewmaster012) [reply zetaclient/evm_client.go:516]: could be non-empty as well.

**Pull request merged**